MapboxCoreNavigation/RouteController.swift

```diff
@@ -351,10 +351,19 @@ extension RouteController: CLLocationManagerDelegate {
         }
 
         // Only check ever 2 minutes for faster route
-        guard location.timestamp.timeIntervalSince(lastLocationDate) >= 120 else { return }
+        guard location.timestamp.timeIntervalSince(lastLocationDate) >= 300 else { return }
         let durationRemaining = routeProgress.durationRemaining
         let currentAlertLevel = routeProgress.currentLegProgress.alertUserLevel
         
+        func compareOldRoute(to route: Route) {
+            // Only use new route if it's at least 10% faster
+            if route.expectedTravelTime <= 0.9 * durationRemaining {
+                // If the upcoming maneuver in the new route is the same as the current upcoming maneuver, don't announce it
+                routeProgress = RouteProgress(route: route, legIndex: 0, alertLevel: currentUpcomingManeuver.maneuverLocation == route.legs[0].steps[1].maneuverLocation ? currentAlertLevel : .none)
```

Comment:
We need to check whether the steps are equivalent, not just whether they occur at the same location. What if the current step turns left at this location, whereas the new route’s first step turns right at the same location? See mapbox/MapboxDirections.swift#151 for a rough sketch of what RouteStep equality would look like.